tools/releases/version.sh

```diff
@@ -1,30 +1,51 @@
 #!/usr/bin/env bash
 
-# Kuma version is built as follows:
+# Kuma version info has the following format:
+# version git-tag git-commit build-date envoy-version
+
+# version is built as follows:
 # 1) If a git tag is present on the current commit, then the version is a git tag (without a starting v)
 # 2) If the branch is "release-X.Y" look at the existing tags and use either X.Y.0-<shortHash> if there's none or
 #     the latest tag with a patch version increased by one (.e.g if latest tag is X.Y.1 the version will be `X.Y.2-<shortHash>`)
 # 3) In non release branch use `0.0.0-$shortHash`
 
+# git-tag - if the current HEAD has a tag then use it otherwise it's "not-tagged"
+# git-commit - HEAD sha
+# build-date - local date if built on CI
+
 # Note: this format must be changed carefully, other scripts depend on it
-exactTag=$(git describe --exact-match --tags 2> /dev/null)
+shortHash=$(git rev-parse --short HEAD 2> /dev/null || echo "no-commit")
```

Comment:
Might be worth doing --short=9 so we always have the same number of characters

Reply:
There are cases in CI where git is not available. We get some weird warning in this case when running any make targets. Could we handle this case as well? (Just set everything to `no-git` for example)

Reply:
added a case for that